src/data_generators/tests/data_design_test.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2021 The Private Cardinality Estimation Framework Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for publisher_data_file.py."""
+
+from absl.testing import absltest
+from tempfile import TemporaryDirectory
+
+from wfa_planning_evaluation_framework.data_generators.publisher_data_file import (
+    PublisherDataFile,
+)
+from wfa_planning_evaluation_framework.data_generators.data_design import DataDesign
+from wfa_planning_evaluation_framework.data_generators.data_set import DataSet
+
+
+class DataDesignTest(absltest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        pdf11 = PublisherDataFile([(1, 0.01), (2, 0.02), (1, 0.04), (3, 0.05)], "pdf11")
+        pdf12 = PublisherDataFile([(2, 0.03), (4, 0.06)], "pdf12")
+        cls.data_set1 = DataSet([pdf11, pdf12], "ds1")
+
+        pdf21 = PublisherDataFile([(1, 0.01), (2, 0.02), (2, 0.04), (3, 0.05)], "pdf21")
+        pdf22 = PublisherDataFile([(2, 0.03), (3, 0.06)], "pdf22")
+        cls.data_set2 = DataSet([pdf21, pdf22], "ds2")
+
+    def test_properties(self):
+        with TemporaryDirectory() as d:
+            dd = DataDesign(d)
```

Comment:
nit: dd = DataDesign(TemporaryDirectory()) and getting rid of the "with" clause can make it more readable since this tests the properties and TemporaryDirectory is not used after initialization.  Up to you though...

Reply:
The reason that I chose not to do that was because the following annoying spurious message is generated:

/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/tempfile.py:819: ResourceWarning: Implicitly cleaning up <TemporaryDirectory '/var/folders/8d/p51l5fk11r50stgb5mvxsgvm00bx10/T/tmpbph2gydi'>
  _warnings.warn(warn_message, ResourceWarning)